### PR TITLE
Restructure headscale documentation

### PR DESCRIPTION
**Pull request by nblock**
Following the spirit of the 0.23 release, this PR aims to restructure the existing documentation to improve
maintanability while providing more information to users. It is mainly a preparation for versioned documentation in the
future which will allow us to evolve the documentation alongside headscale releases. Relevant aspects of this PR:

* Split the documentation into multiple sections:
    * About: Information about the project: FAQ, features, client compat, contributing, …
    * Setup: Requirements, Install page for varous release, Upgrading
    * Usage: "Getting started", Connecting with Android/Apple/Windows
    * Reference: specific documentation (acl, dns, oidc, …) and integration with other components (reverse proxy, Web UI)
* Remove duplication and split the common parts into their own pages
* Document more ways to install headscale (official binaries and packages, distribution, container, …).
* Add a configuration reference
* Reuse existing documentation such as CONTRIBUTING.md
* Replace duplicate content in the README with links to the docs (features, client)
* Update wording to headscale (in regular text), Headscale (beginning of a sentence) and `headscale` when referring to the CLI.
* Add redirects from old links to the new pages

A preview is available here: <https://nblock.github.io/headscale/>

Fixes: #2164
Ref: #1863

---

- [x] have read the [CONTRIBUTING.md](./CONTRIBUTING.md) file
- [x] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [x] updated documentation if needed
- [ ] updated CHANGELOG.md

### CODE REVIEW

File: docs/ref/configuration.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+!!! example "Get the [example configuration from the GitHub repository](https://github.com/juanfont/headscale/blob/main/config-example.yaml)"
+
+    Always select the [same GitHub tag](https://github.com/juanfont/headscale/tags) as the released version you use to
+    ensure you have the correct example configuration. The `main` branch might contain unreleased changes.
+
+    === "View on GitHub"
+
+        * Development version: <https://github.com/juanfont/headscale/blob/main/config-example.yaml>
+        * Version 0.23.0: <https://github.com/juanfont/headscale/blob/v0.23.0/config-example.yaml>
```
Comment (kradalby) [inline docs/ref/configuration.md:19]: Wonder if there is a way these can be variables so we only have to change them in one place 🤔

File: docs/setup/requirements.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- SQLite is used as database.
+- The data directory for headscale (used for private keys, ACLs, SQLite database, …) is located in `/var/lib/headscale`.
+- URLs and values that need to be replaced by the user are either denoted as `<VALUE_TO_CHANGE>` or use placeholder
+  values such as `headscale.example.com`.
+
+Please adjust to your local environment accordingly.
+
+[^1]:
+    The Tailscale client assumes HTTPS on port 443 in certain situations. Serving headscale either via HTTP or via HTTPS
+    on a port other than 443 is possible but sticking with HTTPS on port 443 is strogly recommended for production
```
Comment (kradalby) [inline docs/setup/requirements.md:27]:
```suggestion
    on a port other than 443 is possible but sticking with HTTPS on port 443 is strongly recommended for production
```

File: docs/usage/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+=== "Container"
+
+    ```shell
+    docker exec -it headscale \
+      headscale nodes register --user <USER> --key <YOUR_MACHINE_KEY>
+    ```
+
+### Using a preauthkey
+
+It is also possible to generate a preauthkey and register a node non-interactively. First, generate a preauthkey key on
```
Comment (kradalby) [inline docs/usage/getting-started.md:110]:
```suggestion
It is also possible to generate a preauthkey and register a node non-interactively. First, generate a preauthkey on
```

I think 🤔

**Review by kradalby — APPROVED** [review]
I think this looks great, it looks a lot more coherent and a lot more maintainable than our current documentation. 
This is extremely helpful!

File: docs/ref/configuration.md (hunk 1 above)
Comment (nblock) [reply docs/ref/configuration.md:19]: I have this on my list, could also be useful for admonitions used multiple times. Probably some kind of doc snippets.

Comment (kradalby) [reply docs/ref/configuration.md:19]: Sounds good!

Comment (nblock) [reply docs/ref/configuration.md]: Initial try using the [mkdocs-macros-plugin](https://mkdocs-macros-plugin.readthedocs.io/): https://github.com/juanfont/headscale/pull/2163/commits/9aa60cb45dcffefcb4678de5999119fa67303d83

File: mkdocs.yml [hunk 4]
```diff
@@ -99,6 +100,8 @@ extra:
       link: https://github.com/juanfont/headscale/pkgs/container/headscale
     - icon: fontawesome/brands/discord
       link: https://discord.gg/c84AZQhmpx
+  headscale:
+    version: 0.23.0
```
Comment (kradalby) [inline mkdocs.yml:104]: aweomse

**Comment by kradalby** [comment]
I'm very happy with this, I will let the tests run and then roll out tomorrow unless you have any other things you think should be part of it @nblock

**Comment by nblock** [comment]
> I'm very happy with this, I will let the tests run and then roll out tomorrow unless you have any other things you think should be part of it @nblock

I'm happy with it for now.

**Pull request merged**